core/new-gui/src/app/common/service/notification/notification.service.ts

```diff
@@ -1,53 +1,32 @@
 import { Injectable } from "@angular/core";
-import { Observable, Subject } from "rxjs";
-import { NzMessageDataOptions } from "ng-zorro-antd/message";
-import { HttpErrorResponse } from "@angular/common/http";
-
-export interface Notification {
-  type: "success" | "info" | "error" | "warning" | "loading";
-  message: string;
-  options: NzMessageDataOptions;
-}
+import { NzMessageDataOptions, NzMessageService } from "ng-zorro-antd/message";
 
 /**
  * NotificationService is an entry service for sending notifications
- * to show on NotificationComponent.
  */
 @Injectable({
   providedIn: "root",
 })
 export class NotificationService {
-  private notificationStream = new Subject<Notification>();
-
-  getNotificationStream(): Observable<Notification> {
-    return this.notificationStream.asObservable();
-  }
-
-  sendNotification(notification: Notification) {
-    this.notificationStream.next(notification);
-  }
+  constructor(private message: NzMessageService) {}
 
   success(message: string, options: NzMessageDataOptions = {}) {
-    this.sendNotification({ type: "success", message, options });
+    this.message.success(message, options);
   }
 
   info(message: string, options: NzMessageDataOptions = {}) {
-    this.sendNotification({ type: "info", message, options });
+    this.message.info(message, options);
   }
 
-  error(cause: Error | any, options: NzMessageDataOptions = {}) {
-    this.sendNotification({
-      type: "error",
-      message: cause instanceof Error || cause.hasOwnProperty("message") ? cause.message : cause.toString(),
-      options,
-    });
+  error(message: string, options: NzMessageDataOptions = {}) {
+    this.message.error(message, options);
```

Comment:
I still prefer it to accept an `Error`, that's also a primary reason we want to have this wrapper.

Reply:
I don't think it's a correct design to accept a specific type for a notification service. Notification service should only accept strings, and it's caller's responbility to pass a string in. There is a error type with Property("message"), and the other without Property("message"). There are also many error messages are just string. In a word, I don't think error parsing is part of the Notification service.